CRM/Core/Page/AJAX/Location.php

```diff
@@ -209,71 +209,73 @@ public static function jqCounty() {
   public static function getLocBlock() {
     // i wish i could retrieve loc block info based on loc_block_id,
     // Anyway, lets retrieve an event which has loc_block_id set to 'lbid'.
-    if ($_REQUEST['lbid']) {
-      $params = array('1' => array($_REQUEST['lbid'], 'Integer'));
+    if (!empty($_REQUEST['lbid'])) {
+      $params = ['1' => [$_REQUEST['lbid'], 'Integer']];
       $eventId = CRM_Core_DAO::singleValueQuery('SELECT id FROM civicrm_event WHERE loc_block_id=%1 LIMIT 1', $params);
     }
     // now lets use the event-id obtained above, to retrieve loc block information.
-    if ($eventId) {
-      $params = array('entity_id' => $eventId, 'entity_table' => 'civicrm_event');
+    if (isset($eventId)) {
```

Comment:
I believe `!empty()` is more similar to what was there before.

Reply:
When adding a new location to an event and attempting to use an existing location, $_REQUEST['lbid'] is set to an empty string ("")  which evaluates as TRUE with just an IF ($_REQUEST['lbid']) {}. An "isset" would also evaluate as true.  PHP's "empty" evaluates an empty string ("") as true, that's why I chose to use a negated empty function on the array var.